src/libraries/System.Utf8String.Experimental/src/System/Runtime/Intrinsics/Intrinsics.Shims.cs

```diff
@@ -148,7 +150,86 @@ internal abstract class AdvSimd : ArmBase
         public new abstract class Arm64 : ArmBase.Arm64
```

Comment:
@GrabYourPitchforks, instead of maintaining this `Shims.cs` file, we could just include the `*.PlatformNotSupported.cs` variants of the relevant ISAs: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.PlatformNotSupported.cs

Thoughts? (Note we still need the Shims file for `Vector64/Vector128/Vector256`, but it would avoid needing to update for all of the ISAs)

Reply:
I'm curious about this too. Whenever you select "Go to definition" in Visual Studio, it takes you to the `AdvSimd.PlatformNotSupported.cs` file. I wonder if there's a specific reason for having the same APIs throwing PNSE in two different files.

Reply:
When I tried doing this, VS complained about not understanding the [Intrinsic] attribute. I didn't bother investigating why, but I did see some commented out code in this file with similar [Intrinsic] attributes. Maybe that's the reason @GrabYourPitchforks didn't include the whole file?

Reply:
I have no problem with including the PNSE file as long as we're not introducing public API. We'd need to define a dummy `[Intrinsic]` attribute in the consuming project, but that's very straightforward.

Reply:
Ah, that would be an issue. The types in the PNSE file are public, not internal.

Reply:
I wonder if we should have ifdefs around these, just like how we have ifdefs around some of the new code analysis attributes. It might help code sharing between libraries that target netstandard2.0.
(Really that's an issue for a separate PR, but I wonder if this PR is a forcing function for it.)

Reply:
It actually sounds like we need one for some of the other OOB libraries as well. It might be good to just have an official shim library.

System.Text.Encodings.Web is using ifdefs today, but multitargets netstandard2.0, netcoreapp3.1, and net5; so the ifdefs are going to get a bit crazier

Reply:
For System.Text.Encodings.Web, I resorted to copying this file and extending it as needed. But would be great to have a package for frictionless intrinsics API usage.